File: lib/js/src/view/AppStyles.js
'use strict';

var Colors      = require("../common/styles/Colors.js");
var TypedGlamor = require("typed-glamor/lib/js/src/TypedGlamor.bs.js");

var container = TypedGlamor.css(/* :: */[
      TypedGlamor.background(Colors.background),
      /* :: */[
        TypedGlamor.minHeight(TypedGlamor.vh(100)),
        /* :: */[
          TypedGlamor.display(TypedGlamor.flex),
          /* :: */[
            TypedGlamor.flexDirection(TypedGlamor.column),
            /* :: */[
              TypedGlamor.select("& > footer", /* :: */[
                    TypedGlamor.background(Colors.darkBackground),
                    /* :: */[
                      TypedGlamor.padding3(TypedGlamor.em(1), TypedGlamor.zero, TypedGlamor.em(2)),
                      /* :: */[
                        TypedGlamor.marginTop(TypedGlamor.auto),
                        /* :: */[
                          TypedGlamor.select("& > div", /* :: */[
                                TypedGlamor.display(TypedGlamor.flex),
                                /* :: */[
                                  TypedGlamor.select("& > section", /* :: */[
                                        TypedGlamor.margin2(TypedGlamor.zero, TypedGlamor.em(2)),
                                        /* :: */[
                                          TypedGlamor.opacity(0.5),
                                          /* :: */[
                                            TypedGlamor.transitions(/* :: */[
                                                  /* tuple */[
                                                    TypedGlamor.AnimatableProperty[/* opacity */76],
                                                    TypedGlamor.ms(500),
                                                    TypedGlamor.easeInOut,
                                                    TypedGlamor.ms(0)
                                                  ],
                                                  /* [] */0
                                                ]),
                                            /* :: */[
                                              TypedGlamor.hover(/* :: */[
                                                    TypedGlamor.opacity(1),
                                                    /* [] */0
                                                  ]),
                                              /* :: */[
                                                TypedGlamor.select("& h1", /* :: */[
                                                      TypedGlamor.fontSize(TypedGlamor.em(0.85)),
                                                      /* :: */[
                                                        TypedGlamor.color(TypedGlamor.rgba(255, 255, 255, 0.5)),
                                                        /* :: */[
                                                          TypedGlamor.textTransform(TypedGlamor.lowercase),
                                                          /* :: */[
                                                            TypedGlamor.fontVariant(TypedGlamor.smallCaps),
                                                            /* :: */[
                                                              TypedGlamor.marginBottom(TypedGlamor.em(0.35)),
                                                              /* [] */0
                                                            ]
                                                          ]
                                                        ]
                                                      ]
                                                    ]),
                                                /* :: */[
                                                  TypedGlamor.select("& a", /* :: */[
                                                        TypedGlamor.color(Colors.text),
                                                        /* :: */[
                                                          TypedGlamor.textDecoration(TypedGlamor.none),
                                                          /* :: */[
                                                            TypedGlamor.fontSize(TypedGlamor.rem(0.85)),
                                                            /* :: */[
                                                              TypedGlamor.hover(/* :: */[
                                                                    TypedGlamor.color(TypedGlamor.white),
                                                                    /* [] */0
                                                                  ]),
                                                              /* [] */0
                                                            ]
                                                          ]
                                                        ]
                                                      ]),
                                                  /* [] */0
                                                ]
                                              ]
                                            ]
                                          ]
                                        ]
                                      ]),
                                  /* [] */0
                                ]
                              ]),
                          /* [] */0
                        ]
                      ]
                    ]
                  ]),
              /* [] */0
            ]
          ]
        ]
      ]
    ]);

exports.container = container;
/* container Not a pure module */

File: src/view/AppStyles.ml
open TypedGlamor

let container ~preventScroll = css [
  background Colors.background;

  height (vh 100.);

  display flex;
  flexDirection column;

  select "& > .scroll-container" [
    height (pct 100.);
    display flex;
    flexDirection column;

    if preventScroll then
      overflow hidden
    else
      overflow auto;
  
    select "& > footer" [
      background Colors.darkBackground;

      padding3 ~top:(em 1.) ~h:zero ~bottom:(em 2.);
      marginTop auto;

      select "& > div" [ (* matches the inner width container *)
        display flex;

        select "& > section" [
          margin2 ~v:zero ~h:(em 2.);
          opacity 0.5;
          transitions [AnimatableProperty.opacity, (ms 500), easeInOut, (ms 0)];

          hover [
            opacity 1.;
          ];

          select "& h1" [
            fontSize (em 0.85);
            color (rgba 255 255 255 0.5);
            textTransform lowercase;
            fontVariant smallCaps;
            marginBottom (em 0.35);
          ];

          select "& a" [
            color Colors.text;
            textDecoration none;
            fontSize (rem 0.85);

            hover [
              color white;
            ];
          ];
        ];
      ];
    ];
  ];

  select "& > .mask" [
    position fixed;
    zIndex (int 10);
    offsetTop zero;
    height (vh 100.);
    width (vw 100.);
    background (rgba 0 0 0 0.5)
  ];
]

